lib/hexpm_web/controllers/login_controller.ex

```diff
@@ -3,19 +3,26 @@ defmodule HexpmWeb.LoginController do
 
   plug :nillify_params, ["return"]
 
-  def show(conn, _params) do
+  def show(conn, params) do
     if logged_in?(conn) do
       redirect_return(conn, conn.assigns.current_user, conn.params["return"])
     else
-      render_show(conn)
+      conn
+      |> assign(:token, params["token"])
+      |> render_show()
     end
   end
 
-  def create(conn, %{"username" => username, "password" => password}) do
+  def create(conn, %{"username" => username, "password" => password} = params) do
     case password_auth(username, password) do
       {:ok, user} ->
         breached? = Hexpm.Pwned.password_breached?(password)
-        login(conn, user, password_breached: breached?)
+
+        account_linked? =
+          if params["token"],
+            do: Hexpm.Accounts.Users.link_github_from_token(user, params["token"])
```

Comment:
It looks this expects a boolean return value, but it returns the result of `Repo.transaction`.